Fix:
conftest.c:56:2: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
conftest.c:65:3: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
conftest.c:145:2: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
tty.c:975:6: error: implicit declaration of function 'openpty' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
tty.c:1221:6: error: implicit declaration of function 'login_tty' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
--- configure.orig	2018-01-10 10:29:31.000000000 -0600
+++ configure	2020-11-23 06:04:50.000000000 -0600
@@ -5952,7 +5952,7 @@
 #endif
 int main() {
 	/* exit succesfully if setpgrp() takes two args (*BSD systems) */
-	exit(setpgrp(0, 0) != 0);
+	return setpgrp(0, 0) != 0;
 }
 _ACEOF
 if ac_fn_c_try_run "$LINENO"; then :
@@ -6387,7 +6387,7 @@
 int main() {
   int a = 0;
   isblank(a++);
-  exit(a != 1);
+  return a != 1;
 }
 
 _ACEOF
@@ -8112,7 +8112,7 @@
 	kill((int)getpid(), SIGINT);
 	kill((int)getpid(), SIGINT);
 	/* exit succesfully if don't have to reinstall sighandler when invoked */
-	exit(nsigint != 2);
+	return nsigint != 2;
 }
 _ACEOF
 if ac_fn_c_try_run "$LINENO"; then :
--- joe/tty.c.orig	2018-01-10 10:28:34.000000000 -0600
+++ joe/tty.c	2020-11-23 06:08:05.000000000 -0600
@@ -6,6 +6,7 @@
  *	This file is part of JOE (Joe's Own Editor)
  */
 #include "types.h"
+#include <util.h>
 
 /* Needed for TIOCGWINSZ detection below */
 #ifdef GWINSZ_IN_SYS_IOCTL
